fdbclient/ManagementAPI.actor.cpp

```diff
@@ -2582,22 +2582,38 @@ ACTOR Future<Void> forceRecovery(Reference<IClusterConnectionRecord> clusterFile
 ACTOR Future<UID> auditStorage(Reference<IClusterConnectionRecord> clusterFile,
                                KeyRange range,
                                AuditType type,
+                               double timeoutSeconds,
+                               int retryLimit,
                                bool async) {
 	state Reference<AsyncVar<Optional<ClusterInterface>>> clusterInterface(new AsyncVar<Optional<ClusterInterface>>);
 	state Future<Void> leaderMon = monitorLeader<ClusterInterface>(clusterFile, clusterInterface);
 	TraceEvent(SevDebug, "ManagementAPIAuditStorageBegin");
+	state int retryCount = 0;
 
 	loop {
-		while (!clusterInterface->get().present()) {
-			wait(clusterInterface->onChange());
+		try {
+			while (!clusterInterface->get().present()) {
+				wait(clusterInterface->onChange());
+			}
+			TriggerAuditRequest req(type, range);
+			req.async = async;
+			state UID auditId =
+			    wait(timeoutError(clusterInterface->get().get().triggerAudit.getReply(req), timeoutSeconds));
+			TraceEvent(SevDebug, "ManagementAPIAuditStorageEnd").detail("AuditID", auditId);
+			break;
+		} catch (Error& e) {
+			if (retryCount > retryLimit) {
```

Comment:
I feel it is not necessary to have this retry logic, auditStorage() returns whatever errors it sees, and it should be up to the caller to handle the error, so that we can keep `auditStorage()` simple and clean.

Reply:
Agree. Thanks!